rpc/core/status_test.go

```diff
@@ -0,0 +1,76 @@
+package core_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tendermint/tendermint/p2p"
+	cmtproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/rpc/core"
+)
+
+func TestGetNodeInfo(t *testing.T) {
+	type testCase struct {
+		name            string
+		env             *core.Environment
+		consensusParams cmtproto.ConsensusParams
+		want            uint64
+	}
+	testCases := []testCase{
+		{
+			name: "want 1 when consensus params app version is 1",
+			env:  &core.Environment{P2PTransport: mockTransport{}},
+			consensusParams: cmtproto.ConsensusParams{
+				Version: cmtproto.VersionParams{
+					AppVersion: 1,
+				},
+			},
+			want: 1,
+		},
+		{
+			name: "want 2 if consensus params app version is 2",
+			env:  &core.Environment{P2PTransport: mockTransport{}},
+			consensusParams: cmtproto.ConsensusParams{
+				Version: cmtproto.VersionParams{
+					AppVersion: 2,
+				},
+			},
+			want: 2,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			nodeInfo := core.GetNodeInfo(tc.env, tc.consensusParams)
+			assert.Equal(t, tc.want, nodeInfo.ProtocolVersion.App)
+		})
+	}
+}
+
+// transport is copy + pasted from core because it isn't exported.
```

Comment:
[nit] A link to where this is copied from would be useful for future references.